CommandLine/testing/main.cpp

```diff
@@ -9,9 +9,14 @@ std::string gitMasterDir = "/tmp/enigma-master";
 int main(int argc, char **argv) {
   if (argc == 1) {
     int ret = system(("bash ./ci-regression.sh " + gitMasterDir).c_str());
+
     if (ret != 0) {
-      std::cerr << "Error: ci-regression.sh returned non-zero." << std::endl;
-      return ret;
+      if (WIFEXITED(ret)) {

```

Comment:
I don't know if this is valid for `system()`... this is how we process return codes from our other binaries. Did you read online somewhere that this works, or did you just borrow this code from my test harness class? 😛 I'll assume this is meant to be used this way. But honestly, you could just return 1 instead of mirroring the tool's return code.

Reply:
Sure, I didn't know which was correct, just knew which was "more" correct. Everyone on the internet says this though.
https://stackoverflow.com/a/35318813
http://pubs.opengroup.org/onlinepubs/9699919799/functions/system.html
>If command is not a null pointer, system() shall return the termination status of the command language interpreter in the format specified by waitpid()

Since it's not really defined though, except for POSIX, returning 1 is probably safest.

Reply:
Up to you.

Reply:
Alright, I decided to simply return 1, but return the print value of `system(...)` in the error message. This should make it possibly to safely run this apparatus on Windows and other non-POSIX: dcd9b82f573b5bdbe36380e5e4ac3ac0458d7c96